src/opensynth/data_modules/streaming_data_module.py

```diff
@@ -0,0 +1,182 @@
+# Copyright Contributors to the Opensynth-energy Project.
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+
+import litdata as ld
+import pandas as pd
+import pytorch_lightning as pl
+import torch
+
+from opensynth.data_modules.lcl_data_module import TrainingData
+
+RANDOM_STATE = 0
+g = torch.Generator()
+g.manual_seed(RANDOM_STATE)
+
+
+class StreamData(ld.StreamingDataset):
+    """
+    Streaming Dataset. The dataset should return
+    TrainingData(TypedDict) which contains:
+    - kwh: kWh data
+    - features: Dictionary of features
+
+    To use Faraday on custom datasets, your data module
+    should also return data in the same format.
+
+    For more information on the attributes used for the StreamingDataset, see:
+    https://github.com/Lightning-AI/litdata/blob/main/src/litdata/streaming/dataset.py#L43
+    """
+
+    def __init__(
+        self,
+        data_path: Path,
+        stats_path: Path,
+        storage_options: dict = None,
+        subsample: float = 1.0,
+        shuffle: bool = False,
+        drop_last: bool = True,
+        max_pre_download: int = 2,
+        max_cache_size: str = "100GB",
+    ):
+        """
+        Args:
+            data_path (str): Path to stream data from (can be GCS URI or local path)
+            stats_path (Path): Stats path. Note when loading evaluation
+                dataset, the stats path point to the stats of training data,
+                rather than evaluation data to avoid data leakage!
+            storage_options (dict): Storage options metadata (e.g. GCS project)
+            feature_mean (float): Mean of consumption data (kWh), used for standardisation
+            feature_std (float): STD of consumption data (kWh), used for standardisation
+            shuffle (bool, optional): Whether to shuffle the data - DO NOT ENABLE, DOES NOT GIVE GOOD RESULTS
+            drop_last (bool, optional): Drop last items so processes return same amount of data
+        """
+        super().__init__(
+            input_dir=data_path,
+            storage_options=storage_options,
+            shuffle=shuffle,
+            drop_last=drop_last,
+            subsample=subsample,
+            max_pre_download=max_pre_download,
+            max_cache_size=max_cache_size,
+        )
+        # Parse stats
+        self.df_stats = pd.read_csv(stats_path)
```

Comment:
If we're providing a `data_path` explicitly, I think we're an assumption here that this file is already randomly shuffled and we're only loading 1 of the many many .csvs.

If so we should call this out in the doc strings.

Alternatively, have the `mean` and `std` explicitly provided into the dataset, i.e. user has to precalculate this, rather than inferred from a data file.